src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -0,0 +1,95 @@
+(ns puppetlabs.jdbc-util.pool
+  (:require [clojure.set :as set]
+            [clojure.tools.logging :as log])
+  (:import [com.zaxxer.hikari HikariConfig HikariDataSource]
+           java.io.Closeable
+           java.sql.SQLTransientConnectionException
+           javax.sql.DataSource))
+
+(defn connection-pool
+  [^HikariConfig config]
+  (let [ds (HikariDataSource. config)]
+    {:datasource ds}))
+
+(defn- set-option
+  [^HikariConfig config option value]
+  (case option
+    :username (.setUsername config value)
+    :password (.setPassword config value)
+    :data-source-class-name (.setDataSourceClassName config value)
+    :jdbc-url (.setJdbcUrl config value)
+    :driver-class-name (.setDriverClassName config value)
+    :auto-commit (.setAutoCommit config value)
+    :connection-timeout (.setConnectionTimeout config value)
+    :idle-timeout (.setIdleTimeout config value)
+    :max-lifetime (.setMaxLifetime config value)
+    :connection-test-query (.setConnectionTestQuery config value)
+    :minimum-idle (.setMinimumIdle config value)
+    :maximum-pool-size (.setMaximumPoolSize config value)
+    :metric-registry (.setMetricRegistry config value)
+    :health-check-registry (.setHealthCheckRegistry config value)
+    :pool-name (.setPoolName config value)
+    (throw (IllegalArgumentException. (format "%s is not a recognized HikariCP option" (str option))))))
+
+(defn options->hikari-config
```

Comment:
There may be value to schematizing this, but I kind of think it makes more sense to do at the config parsing layer (which this library could provide helpers for). The call to validate checks most of these anyway, and runs no risk of diverging from hikari.